src/Analyzers/Orchestration/OrchestrationAnalyzer.cs

```diff
@@ -0,0 +1,285 @@
+
+                // there are 8 AddOrchestratorFunc overloads
+                if (invocation.TargetMethod.Name != "AddOrchestratorFunc")
+                {
+                    return;
+                }
+
+                // all overloads have the parameter orchestrator, either as an Action or a Func
+                IArgumentOperation orchestratorArgument = invocation.Arguments.First(a => a.Parameter!.Name == "orchestrator");
+                if (orchestratorArgument.Value is not IDelegateCreationOperation delegateCreationOperation)
+                {
+                    return;
+                }
+
+                IMethodSymbol? methodSymbol = null;
+                switch (delegateCreationOperation.Target)
+                {
+                    case IAnonymousFunctionOperation lambdaOperation:
+                        // use the containing symbol of the lambda (e.g. the class declaring it) as the method symbol
+                        methodSymbol = ctx.ContainingSymbol as IMethodSymbol;
+                        break;
+                    case IMethodReferenceOperation methodReferenceOperation:
+                        // use the method reference as the method symbol
+                        methodSymbol = methodReferenceOperation.Method;
+                        break;
+                    default:
+                        break;
+                }
+
+                if (methodSymbol == null)
+                {
+                    return;
+                }
+
+                // try to get the name of the orchestration from the method call, otherwise use the containing type name
+                IArgumentOperation nameArgument = invocation.Arguments.First(a => a.Parameter!.Name == "name");
+                Optional<object?> name = nameArgument.GetConstantValueFromAttribute(ctx.Operation.SemanticModel!, ctx.CancellationToken);
+                string orchestrationName = name.Value?.ToString() ?? methodSymbol.Name;
+
+                AnalyzedOrchestration orchestration = new(orchestrationName);
+
+                SyntaxNode funcRootSyntax = delegateCreationOperation.Syntax;
+
+                FindInvokedMethods(ctx.Operation.SemanticModel!, funcRootSyntax, methodSymbol, orchestration, result);
+            },
+                OperationKind.Invocation);
+
+            // allows concrete implementations to register specific actions/analysis and then check if they happen in any of the orchestration methods
+            this.RegisterAdditionalCompilationStartAction(context, result);
+        });
+    }
+
+    /// <summary>
+    /// Register additional actions to be executed after the compilation has started.
+    /// It is expected from a concrete implementation of <see cref="OrchestrationAnalyzer"/> to register a
+    /// <see cref="CompilationStartAnalysisContext.RegisterCompilationEndAction"/>
+    /// and then compare that any discovered violations happened in any of the symbols in <paramref name="orchestrationAnalysisResult"/>.
```

Comment:
```suggestion
    /// and then compare that to any discovered violations happened in any of the symbols in <paramref name="orchestrationAnalysisResult"/>.
```

Let me know if you mean something different!